### DISCUSSION TITLE
Purpose of multiple_images ?

### DISCUSSION THREAD

**Discussion by gateway** [rkhamilton/vqgan-clip-generator#22] [Q&A] [answered]
So I'm testing as usual so it may not be 100% clear to me what the function multiple_images does outside of saving multiple images.

Im testing something like this.. 

```import vqgan_clip.generate
from vqgan_clip.engine import VQGAN_CLIP_Config
import os

config = VQGAN_CLIP_Config()
config.output_image_size = [900,506]
#config.output_image_size = [506,900]
config.init_weight = 1.0
config.init_image = 'images/bitcoin.jpg'
config.init_noise = 'gradient'
text_prompts = 'a crypto currency made out of liquide metal mathmatical equations and computer circuit board volumetric lighting VRay Unreal Engine'

vqgan_clip.generate.multiple_images(eng_config = config,
        text_prompts = text_prompts,
        iterations = 10,
        save_every = 5,
        num_images_to_generate=60)
```

What ends up is a bunch of frames that look pretty similar to each other with slight variations.

![image](https://user-images.githubusercontent.com/519563/138367294-d3c623d5-bfa6-46a6-8f22-6886f5450602.png)

My thought would be that this would slowly generate a bunch of images that I could set a iteration goal for it to go to over x saves.

Meaning If I wanted the end result of the multiple images to reach a iteration value of 500 over 60 images to generate and save every 5 that it might be saving every 5 iterations on its way to reaching the 500 value.

500 iterations over 60 images = 8.33333 lets say 8

image 1 (8 iterations)
image 2 (16 iterations) 
...

My manual way but more something along these lines..

![image](https://user-images.githubusercontent.com/519563/138368180-0c89f7bb-a5ab-41e0-a42e-533c1f653956.png)

? :)

**Comment by rkhamilton** [comment, marked as answer]
Please take a look at the docstring for the functions in generate. If you are using something like VS Code, you can hover your mouse over the function name and it should explain what it's purpose is. For example, with multiple_images you would see the help below.
![image](https://user-images.githubusercontent.com/8406344/138368734-606923f1-f763-491f-be27-67a647a3be6e.png)

Re: multiple_images, you are correct. It just generated multiple single independent images. A friend of mine was creating images with no initial image, and they varied wildly based on the random seed. He would run the script repeatedly and fish through the results for the more interesting images. The multiple_images script facilitates that process. It's not any faster than looping over single_image, and may not be that useful.

For your use case of building toward the same prompt and saving as you go is what the video_frames function does. As does the zoom_video_frames function. Set your iterations to 500, save_every to 8, and you will get what you were looking for.

**Reply by gateway** [reply]
ahh nice.. video_frames is my new friend :) 

![image](https://user-images.githubusercontent.com/519563/138371790-83f4a4ad-2dc4-4c5c-a353-75abbbc4f0e1.png)

**Reply by rkhamilton** [reply]
I’m curious, what is your use case? I find video_frames to be not much use because the image stabilizes after 100-200 iterations  unless you are changing prompts. With zoom_video_frames the image evolves forever so I tend to use that and rarely touch video_frames.

**Reply by gateway** [reply]
Yea I noticed it doesn't change much after 200 give or take.. I just like the evolution to be able to see.  I think ideally I want to do this but feed in a text file of different prompts and scan to see which one I like better with what text prompt and iteration..  I love to see things in a way where I can visualize the evolution similar to this..
![bmlogo](https://user-images.githubusercontent.com/519563/138380087-48b9e640-44d2-4762-866d-13ae0286cdde.jpg)
